src/BlueprintParser.h

```diff
@@ -357,14 +357,37 @@ namespace snowcrash {
                 return;
             }
 
-            // Otherwise, add the respective entries to the tables
-            if (typeDefinition.typeSpecification.name.base != mson::UndefinedTypeName) {
+            mson::BaseTypeName baseTypeName = typeDefinition.typeSpecification.name.base;
+
+            // Initialize an entry in the dependency table if not found
+            if (pd.namedTypeDependencyTable.find(identifier) != pd.namedTypeDependencyTable.end()) {
```

Comment:
Yeah, I agree this needs to be split into multiple functions. This is only 100 lines, am I missing something else?